mungegithub/mungers/issue-categorizer.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"strings"
+	"net/http"
+	"net/url"
+
+	"k8s.io/contrib/mungegithub/github"
+
+	"github.com/spf13/cobra"
+	"github.com/golang/glog"
+	"io/ioutil"
+	"k8s.io/contrib/mungegithub/features"
+)
+// SizeMunger will update a label on a PR based on how many lines are changed.
+// It will exclude certain files in it's calculations based on the config
+// file provided in --generated-files-config
+type LabelMunger struct {
+}
+
+func (LabelMunger) Initialize(config *github.Config, features *features.Features) error {
+	return nil
+}
+
+func (LabelMunger) Name() string { return "LabelMunger" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (LabelMunger) RequiredFeatures() []string { return []string{} }
+
+func (LabelMunger) AddFlags(cmd *cobra.Command, config *github.Config) {
+}
+
+func init() {
+	s := &LabelMunger{}
+	RegisterMungerOrDie(s)
+}
+
+// EachLoop is called at the start of every munge loop
+func (LabelMunger) EachLoop() error { return nil }
+
+
+// Munge is the workhorse the will actually make updates to the PR
+func (s *LabelMunger) Munge(obj *github.MungeObject) {
+	//this munger only works on issues
+	if obj.IsPR() {
+		return
+	}
+
+
+
+	issue := obj.Issue
+
+	existingLabels := issue.Labels
+	for _, label := range existingLabels{
+		//if a component or team label has already been added do not re-add it
+		if strings.HasPrefix(label, "component") || strings.HasPrefix(label, "team"){
+			return
+		}
+	}
+
+
+
+	routingLabelsToApply, err := http.PostForm("http://localhost:5000",
+		url.Values{"title": {*issue.Title}, "body": {*issue.Body}})
+
+	if routingLabelsToApply.StatusCode != 200{
+		glog.Error("Not 200 status code: " + string(routingLabelsToApply.StatusCode))
```

Comment:
the line below this also checks if `err != nil`.   Is that redundant.  Does a non 200 status code automatically indicate there was an error and can I removed the second check.